File: DynamicProgramming1/PadovanSequence/CPPPadovanSequence/PadovanSequence.cpp
#include <iostream>
#include <vector>

#define MAX 100

using namespace std;

vector<long long> padovan_sequence(MAX + 1);

void get_padovan_sequence()
{
	padovan_sequence.push_back(0);
	padovan_sequence.push_back(1);
	padovan_sequence.push_back(1);
	padovan_sequence.push_back(1);

	for (int i = 4; i <= MAX; i++)
		padovan_sequence.push_back(padovan_sequence[i - 2] + padovan_sequence[i - 3]);
}

int main()
{
	int t, n;

	get_padovan_sequence();

	cin >> t;

	for (int i = 0; i < t; i++)
	{
		cin >> n;
		cout << padovan_sequence[n] << "\n";
	}

	return 0;
}

File: DynamicProgramming1/PadovanSequence/GoPadovanSequence/PadovanSequence.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

const (
	max = 100
)

var (
	padovanSequence [max + 1]int64
)

func getPadovanSequence() {
	padovanSequence[0] = 0
	padovanSequence[1] = 1
	padovanSequence[2] = 1
	padovanSequence[3] = 1

	for i := 4; i <= max; i++ {
		padovanSequence[i] = padovanSequence[i-2] + padovanSequence[i-3]
	}
}

func main() {
	br := bufio.NewReader(os.Stdin)
	bw := bufio.NewWriter(os.Stdout)

	getPadovanSequence()

	t := 0
	fmt.Fscanln(br, &t)

	for i := 0; i < t; i++ {
		n := 0
		fmt.Fscanln(br, &n)
		fmt.Fprintln(bw, padovanSequence[n])
	}

	bw.Flush()
}
